Refactored building tools for Ant-based tests

dmatej opened the pull request:
* Fixed closing resources
* reportbuilder and reporter moved to the standard Maven lifecycle
   * Before their usage by Ant scripts execute a special local maven project which collects dependencies for tests - it is more visible than using Ant's copy task and also it is less work to add/remove something.* Removed binaries from the lib directory of Ant tests
* Some Ant tasks were reimplemented - created custom HTTP clients able to send authentication requests (original used Tomcat's GTask from 2001, which doesn't have successor).
* Some tests were moved to application-tests, executed by maven. 
   * I will probably continue in another PR with the rest of security_all tests soon

dmatej committed:
Using target instead of tasks

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Refactored closing streams in report builder

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

pzygielo commented:
Is that necessary to have `appserver/tests/appserv-tests/lib/reportbuilder.jar` checked-in?

dmatej commented:
> Is that necessary to have `appserver/tests/appserv-tests/lib/reportbuilder.jar` checked-in?

I had the same question, but I have to admit that for now I resigned. It is small and the module stands outside the standard reactor for some reason. It seems it is directly used by Ant based tests, so I would have to change it also in Jenkinsfile, etc. 

But perhaps I should try to change it in this PR, it should be doable ...

dmatej committed:
The report-builder moved to standard Maven reactor

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The reporter moved to standard Maven reactor too

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed resolving GF_VERSION property value

- the root pom.xml is not archived from main maven build.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Dependencies moved to the main reactor + executable from gftest.sh on jenkins

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed paths to reporter.jar

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed naming_all tests

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Enabled staging profile in several Ant tests + fixed dependency

dmatej committed:
Removed logging binaries from Ant tests, replaced by maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
junit and testng moved to maven downloader too

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The webtest.jar moved to maven downloader too

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed unused or redundant files fron Ant ZombieTest Army

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed NPE

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
JMAX http servlet test moved to Maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The jmac httpservletchallenge moved to maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed log message

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed httpServletChallenge

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
HttpServletChallengeAuthTest moved from Ant to Maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The httpsoapembedded test moved to maven

- basically rewritten, because Ant allowed lifecycle like:
  compile-deploy-download-generate-compile-deploy-test
  - but that is not possible with Maven and Shrinkwrap. So I created wsdl and
    xsd and used maven plugin to generate required classes first.


Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added log to StandardWrapper

- when servlet-based soap service crashed, thrown exception was not logged
  anywhere

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed log message

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

# Conflicts:
#	appserver/security/webservices.security/src/main/java/com/sun/enterprise/security/webservices/CommonServerSecurityTube.java
#	appserver/security/webservices.security/src/main/java/com/sun/enterprise/security/webservices/server/ServerSecurityPipe.java

dmatej committed:
Updated version number

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed missing resource bundle warning in logs

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed several unused targets from security_all

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added support for HTTPS and keytool to tests

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
JMAC HTTPS test moved to application tests, disabled

- failed in Ant that is why it was probably excluded
- fails here too, but I think it detected some bug

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed unused targets and properties from jdbc_all

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
HttpMethodAuthTest moved to maven + HTTP1.0 client supporting custom method

- that was originally implemented by the org.apache.tomcat.task.GTest class
  from Tomcat webserver 3.2 of 2001.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The defaultp2r test moved to maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The jsp2sful tests moved to Maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The multirolemapping tests moved to maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The SimpleMultiRoleMappingTest moved from ant to maven

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Temporal solution - s1asCN property was already removed

- it also expected that the server certificate was created with the same CN,
  which wasn't always true, so the previous was not more reliable than this.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed ECL Emma from few ant tests, the obsoleted code coverage library

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed IDEA project files from Ant tests

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/ee/jmac/MessageSecurityConfigEventListenerImpl.java:
```diff
@@ -86,7 +86,10 @@ public <T extends ConfigBeanProxy> NotProcessed changed(TYPE type, Class<T> chan
 
     private <T extends ConfigBeanProxy> NotProcessed handle(T instance) {
         if (instance instanceof MessageSecurityConfig) {
-            AuthConfigFactory.getFactory().refresh();
```
Fixed NPE

dmatej commented on appserver/security/webservices.security/src/main/java/com/sun/enterprise/security/webservices/server/ServerSecurityPipe.java:
```diff
@@ -157,6 +157,7 @@ private Packet processRequest(Packet request) throws Exception {
             authenticationService.auditInvocation(validatedRequest, status);
         }
 
+        _logger.log(FINE, "Validation status: {0}", status);
```
I moved the log and fixed the log message - the original message key did not exist.

dmatej commented on appserver/tests/application/pom.xml:
```diff
@@ -122,6 +149,29 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
The original Ant based impl executed wsgen, while in maven it would lead to a chicken - egg problem. So I created WSDL files, used the plugin to generate required classes which were then available for the test.

dmatej commented on appserver/tests/application/src/test/java/org/glassfish/main/test/app/security/jmac/https/JmacHttpsTest.java:
```diff
@@ -0,0 +1,203 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.stringContainsInOrder;
+
+@Disabled("Fails with GF 7.0.11, the attribute jakarta.servlet.request.X509Certificate is not set for some reason.")
```
This test was failing in Ant too, I have found that the attribute can be set by Catalina or by Grizzly, but it is not set. I have no idea why, it seems as a bug. I transformed the test to Maven, so now it should be easier to reproduce and fix it.
I remember we extended this feature in Payara, so it was possible to use even parts of the DN, maybe we were fixing yet something before, but I don't remember what.
